packages/framework/memo-cache/README.md

```diff
@@ -0,0 +1,141 @@
+
+### When to Use This
+
+This package can be beneficial in two scenarios:
+
+#### Performance
+
+If the routine to be memoized is expensive, then caching the results can boost performance. Note that cache lookups have cost themselves so memoizing a trivial function will likely be slower.
+
+Also note that every additional key adds a level of depth to the hierarchical cache. This has expense and reduces the likelihood of data being already in the cache. Collapsing the inputs to a manageable set helps optimize this. For instance, if building a style from a theme definition pulls 8 values from a theme, it is more efficient to key the resulting object on the theme, than to key each property individually.
+
+#### Object Identity
+
+The other benefit to this pattern is maintaining object identity between subsequent calls. In react-native the object identity of the style property will sometimes be compared, even if the values within are identical, the shallow props compare will not see the objects as the same.
+
+Similarly if a style is being turned into a CSS class (which is expensive), a `WeakMap` to map style objects to CSS classes will only work if the object identities are maintained.
+
+## Usage guide
+
+The baseline cache pattern is defined by the following type:
+
+```ts
+export type GetMemoValue<T, TGet = any> = (
+  factory: T | () => T,
+  keys: any[]
+) => [T, GetMemoValue<TGet>];
+```
+
+The parameters are used as follows:
+
+| Param     | Description                                                                                                                                                                                                                                                                             |
+| --------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `factory` | This is typically a function, often a simple closure, which returns a value. This value will be cached for a given set of keys. Subsequent calls will just return the value without executing the function. This can also be a value type in which case that will be returned directly. |
+| `keys`    | Variable parameter list, used as the keys for caching                                                                                                                                                                                                                                   |
```

Comment:
might be good to note that the ordering of keys is important here, at least as I've understood the reduce code in `getCacheEntry`